README.md

```diff
@@ -17,98 +16,97 @@ gem 'nxt_http_client'
 
 And then execute:
 
-    $ bundle
-
-Or install it yourself as:
-
-    $ gem install nxt_http_client
+```sh
+bundle
+````
 
 ## Usage
 
-A typical client could look something like this:
+Here's a simple HTTP client with this gem:
 
 ```ruby
-class UserFetcher < Client
-  def initialize(id)
-    @url = ".../users/#{id}"
-  end
-
-  def call
-    get(url) do |response_handler|
-      response_handler.on(:success) do |response|
-        JSON(response.body)
-      end
-    end
+client = NxtHttpClient::Client.make do 
+  configure do |config|
+    config.base_url = 'www.httpstat.us'
+    config.request_options.deep_merge!(
+      headers: { API_KEY: '1993' },
+      followlocation: true
+    )
   end
-
-  private
-
-  attr_reader :url
 end
+
+client.get('200')
 ```
 
-In order to setup a shared configuration you would therefore setup a client base class. The configuration and any
-response handler or callbacks you setup in your base class are then inherited to your concrete client implementations.
+This is good when you need a one-off client to make some quick API calls. 
+However, you can go further by creating custom client classes with shared configuration,
+and customizing them as needed.
+For example, you could have a base client for interacting with a specific service:
```

Comment:
```suggestion
However, if you need a simple HTTP client for some ad-hoc one-off task, you can configure one using `.make`, without necessity to define a separate client class. Please note that this method should not be used for any clients that are actively being utilised in the business logic.
```

And swap this with the paragraph and code snippet above.

Reply:
Done. Swapped the examples to favour the class one.

Reply:
Thank you!